### PR TITLE
Apply JAVA_HOME for local template-based SAM debug sessions

### PR DESCRIPTION
**Pull request by rli**
#2493

## License
I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: jetbrains-core/tst/software/aws/toolkits/jetbrains/services/lambda/execution/sam/JavaSamInvokeRunnerTest.kt [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+        runInEdtAndWait {
+            runWriteAction {
+                ProjectJdkTable.getInstance().addJdk(sdk, projectRule.fixture.projectDisposable)
+                ProjectRootManager.getInstance(projectRule.project).projectSdk = sdk
+            }
+        }
+    }
+
+    @Test
+    fun addsJavaHomeEnvVarForJavaTemplate() {
```
Comment (hunterwerlla) [inline jetbrains-core/tst/software/aws/toolkits/jetbrains/services/lambda/execution/sam/JavaSamInvokeRunnerTest.kt:51]: Shouldn't there be a test for handler based in here as well?

File: jetbrains-core/tst/software/aws/toolkits/jetbrains/utils/RunConfigTestUtils.kt [hunk 2]
```diff
@@ -75,6 +77,20 @@ fun executeRunConfiguration(runConfiguration: RunConfiguration, executorId: Stri
     return executionFuture
 }
 
+fun getState(runConfiguration: RunConfiguration, executorId: String = DefaultRunExecutor.EXECUTOR_ID): RunProfileState? {
```
Comment (hunterwerlla) [inline jetbrains-core/tst/software/aws/toolkits/jetbrains/utils/RunConfigTestUtils.kt:80]: I didn't realize we had 3 versions of this, nice cleanup

File: jetbrains-core/src/software/aws/toolkits/jetbrains/services/lambda/upload/steps/BuildLambda.kt [hunk 3]
```diff
@@ -12,25 +12,29 @@ import software.aws.toolkits.jetbrains.utils.execution.steps.MessageEmitter
...
+    val samOptions: SamOptions
+)
+
 class BuildLambda(
-    private val templatePath: Path,
-    private val logicalId: String? = null,
-    private val buildDir: Path,
-    private val buildEnvVars: Map<String, String> = emptyMap(),
-    private val samOptions: SamOptions
+    private val request: BuildLambdaRequest
```
Comment (hunterwerlla) [inline jetbrains-core/src/software/aws/toolkits/jetbrains/services/lambda/upload/steps/BuildLambda.kt:24]: super minor nit, reformat to 1 line?

File: jetbrains-core/tst/software/aws/toolkits/jetbrains/services/lambda/execution/sam/JavaSamInvokeRunnerTest.kt (hunk 1 above)
Comment (rli) [reply jetbrains-core/tst/software/aws/toolkits/jetbrains/services/lambda/execution/sam/JavaSamInvokeRunnerTest.kt]: added

**Review by hunterwerlla — APPROVED** [review]

File: jetbrains-core/src/software/aws/toolkits/jetbrains/services/lambda/execution/sam/SamRunningState.kt [hunk 4]
```diff
@@ -84,35 +85,16 @@ class SamRunningState(
...
         samState.pathMappings = createPathMappings(lambdaBuilder, lambdaSettings, buildLambdaRequest)
         val buildWorkflow = buildWorkflow(environment, settings, samState, buildLambdaRequest, buildView)
 
         return DefaultExecutionResult(buildView, buildWorkflow)
     }
 
-    private fun createPathMappings(lambdaBuilder: LambdaBuilder, settings: LocalLambdaRunSettings, buildRequest: BuildRequest): List<PathMapping> {
-        val defaultPathMappings = lambdaBuilder.defaultPathMappings(buildRequest.template, buildRequest.logicalId, buildRequest.buildDir)
+    private fun createPathMappings(lambdaBuilder: LambdaBuilder, settings: LocalLambdaRunSettings, buildRequest: BuildLambdaRequest): List<PathMapping> {
+        val defaultPathMappings = lambdaBuilder.defaultPathMappings(buildRequest.templatePath, buildRequest.logicalId ?: dummyLogicalId, buildRequest.buildDir)
```
Comment (abrooksv) [inline jetbrains-core/src/software/aws/toolkits/jetbrains/services/lambda/execution/sam/SamRunningState.kt:97]: When is buildRequest.logicalId null?

**Review by abrooksv — APPROVED** [review]

File: jetbrains-core/src/software/aws/toolkits/jetbrains/services/lambda/execution/sam/SamRunningState.kt (hunk 4 above)
Comment (rli) [reply jetbrains-core/src/software/aws/toolkits/jetbrains/services/lambda/execution/sam/SamRunningState.kt:97]: it's nullable through code in LambdaWorkflows.kt

Comment (abrooksv) [reply jetbrains-core/src/software/aws/toolkits/jetbrains/services/lambda/execution/sam/SamRunningState.kt:97]: yaaaa I saw that later :(

Is it worth making a not null version that extends the BuildLambda? BuildLambdaWithId?

**Pull request merged**